warp10/src/main/java/io/warp10/script/functions/TOGTS.java

```diff
@@ -127,47 +131,68 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     // if there is no type map, the output is a map of GTS (v2.3.0 signature), or a map of lists of GTS
     if (null == typeMap) {
       Map<String, ArrayList<GeoTimeSerie>> result = new HashMap<String, ArrayList<GeoTimeSerie>>();
+      GeoTimeSerie[] series = new GeoTimeSerie[5];
+
       for (GTSDecoder decoder: decodersInput) {
-        Map<String, GeoTimeSerie> series = new HashMap<String, GeoTimeSerie>();
+        Arrays.fill(series, null);
+        long remaining = decoder.getCount();
         GeoTimeSerie gts;
         while (decoder.next()) {
           Object value = decoder.getBinaryValue();
+          remaining--;
 
-          String type = "DOUBLE";
+          int typeidx = 0; // LONG
```

Comment:
don't need to be initialized.

Reply:
Always better to initialize explicitely, won't fix.

Reply:
OK, can close.